cudax/include/cuda/experimental/__stf/utility/core.cuh

```diff
@@ -218,6 +218,7 @@ template <typename T>
 auto each(T to)
 {
   static_assert(!::std::is_pointer_v<T>, "Use the two arguments version of each() with pointers.");
+  assert(to >= 0 && "Attempt to iterate from 0 to a negative value.");
```

Comment:
This will give a ton of warnings
```suggestion
  if constexpr(_CCCL_TRAIT(is_signed, T)) {
    assert(to >= 0 && "Attempt to iterate from 0 to a negative value.");
  }
```